python/IECoreDelightTest/RendererTest.py

```diff
@@ -609,6 +650,458 @@ def testOptions( self ) :
+			],
+			output = "splineHandle"
+		)
+
+		o = r.object(
+			"testPlane",
+			IECoreScene.MeshPrimitive.createPlane( imath.Box2f( imath.V2f( -1 ), imath.V2f( 1 ) ) ),
+			r.attributes( IECore.CompoundObject( { "osl:surface" : network } ) )
+		)
+		del o
+
+		r.render()
+		del r
+
+		nsi = self.__parseDict( self.temporaryDirectory() / "test.nsi" )
+
+		shaders = { k: v for k, v in nsi.items() if nsi[k]["nodeType"] == "shader" }
+		self.assertEqual( len( shaders ), 1 )
+		shader = shaders[next( iter( shaders ) )]
+
+		self.assertEqual( shader["splinePositions"], [ 0, 0, 0, 1, 1, 1 ] )
+		self.assertEqual(
+			shader["splineValues"],
+			[
+				imath.Color3f( 1, 0, 0 ),
+				imath.Color3f( 1, 0, 0 ),
+				imath.Color3f( 1, 0, 0 ),
+				imath.Color3f( 0, 0, 1 ),
+				imath.Color3f( 0, 0, 1 ),
+				imath.Color3f( 0, 0, 1 )
+			]
+		)
+		self.assertEqual( shader["splineBasis"], "linear" )
+
+	# Helper methods used to check that NSI files we write contain what we
+	# expect. The 3delight API only allows values to be set, not queried,
+	# so we build a simple dictionary-based node graph for now.
+
+	def __connectionSource( self, connection, nsi ) :
+
+		cSplit = connection.split( '.' )
+		if len( cSplit ) == 0 :
+			return None
+
+		if len( cSplit ) == 1 :
+			return nsi[cSplit[0][1:-1]]  # remove <>
+
+		return nsi[cSplit[0][1:-1]][cSplit[1]]
+
+	def __parseDict( self, nsiFile ) :
+
+		reArraySplit = re.compile( r'(?P<varType>.*)\[(?P<arrayLength>[0-9]+)\]' )
+
+		root = {
+			".root": { "nodeType": "root", "objects": [], "geometryattributes": [], },
+			".global": { "nodeType": "global", },
+		}
+		tokens = deque()
+		with open( nsiFile, encoding = "utf-8" ) as f :
+			for i in f.readlines() :
+				if not i.startswith( '#' ) :
+					tokens += shlex.split( i )
+
+		currentNode = None  # The node to add attributes to
+		currentTime = None # The time to add attributes to
+
+		while len( tokens ) :
+			token = tokens.popleft()
+			if token == "Create" :
+				node = tokens.popleft()
+				root.setdefault( node, {} )["nodeType"] = tokens.popleft()
+			elif token == "SetAttribute" :
+				currentNode = tokens.popleft()
+				currentTime = None
+			elif token == "Connect" :
+				sourceNode = tokens.popleft()
+				sourceAttr = tokens.popleft()
+				destNode = tokens.popleft()
+				destAttr = tokens.popleft()
+
+				source = "<{}>".format( sourceNode ) + ( ( "." + sourceAttr ) if sourceAttr != "" else "" )
+
+				root[destNode].setdefault( destAttr, [] ).append( source )
+
+			elif token == "SetAttributeAtTime" :
+				currentNode = tokens.popleft()
+				currentTime = float( tokens.popleft() )
+			elif token == "Connect" :
+				sourceNode = tokens.popleft()
+				sourceAttr = tokens.popleft()
+				destNode = tokens.popleft()
+				destAttr = tokens.popleft()
+
+				source = "<{}>".format( sourceNode ) + ( ( "." + sourceAttr ) if sourceAttr != "" else "" )
+
+				root[destNode].setdefault( destAttr, [] ).append( source )
```

Comment:
This is a duplicate of the one above - I don't think it's reachable.

Reply:
Good catch, that was an oversight in the midst of squashing and keeping things separated. Fixed in a combination of https://github.com/GafferHQ/gaffer/pull/5432/commits/27a013ed72782e6050396c0138e5a59abc05e610 and https://github.com/GafferHQ/gaffer/pull/5432/commits/5d5e38f8ac6b1dd4b507fbf73fc8bdedaee3dde2.